flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/TypeInferenceUtil.java

```diff
@@ -209,6 +209,47 @@ public static TableException createUnexpectedException(
                 cause);
     }
 
+    /**
+     * Validates argument counts.
+     *
+     * @param argumentCount expected argument count
+     * @param actualCount actual argument count
+     * @param throwOnFailure boolean indicating whether a ValidationException must be thrown or not
```

Comment:
how about we're a bit more specific?
```suggestion
     * @param throwOnFailure if true, the function throws a {@link ValidationException} if the actual value does not meet the expected argument count
```

Reply:
Applied suggestion